api/api-admins-config-v1/src/main/java/com/thoughtworks/go/apiv1/adminsconfig/AdminControllerV1Delegate.java

```diff
@@ -0,0 +1,128 @@
+import com.thoughtworks.go.api.CrudController;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.apiv1.adminsconfig.representers.AdminsRepresenter;
+import com.thoughtworks.go.config.AdminsConfig;
+import com.thoughtworks.go.server.newsecurity.utils.SessionUtils;
+import com.thoughtworks.go.server.service.AdminsConfigService;
+import com.thoughtworks.go.server.service.EntityHashingService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.spark.Routes;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+
+import static com.thoughtworks.go.api.util.HaltApiResponses.haltBecauseEtagDoesNotMatch;
+import static spark.Spark.*;
+
+public class AdminControllerV1Delegate extends ApiController implements CrudController<AdminsConfig> {
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private final EntityHashingService entityHashingService;
+    private final AdminsConfigService adminsConfigService;
+
+
+    public AdminControllerV1Delegate(ApiAuthenticationHelper apiAuthenticationHelper,
+                                     EntityHashingService entityHashingService, AdminsConfigService service) {
+        super(ApiVersion.v1);
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.entityHashingService = entityHashingService;
+        this.adminsConfigService = service;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerPath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+            before("", this::verifyContentType);
+            before("/*", this::verifyContentType);
+            before("", mimeType, apiAuthenticationHelper::checkAdminUserAnd403);
+            before("/*", mimeType, apiAuthenticationHelper::checkAdminUserAnd403);
+
+            get("", mimeType, this::show);
+            put("", mimeType, this::update);
+        });
+    }
+
+    public String show(Request req, Response res) throws IOException {
+        AdminsConfig adminsConf = adminsConfigService.systemAdmins();
+        if (isGetOrHeadRequestFresh(req, adminsConf)) {
+            return notModified(res);
+        } else {
+            setEtagHeader(adminsConf, res);
+            return writerForTopLevelObject(req, res, writer -> AdminsRepresenter.toJSON(writer, adminsConf));
+        }
+    }
+
+    public String update(Request req, Response res) throws IOException {
+        AdminsConfig adminsConfigFromRequest = getEntityFromRequestBody(req);
+        AdminsConfig adminsConfigFromServer = adminsConfigService.systemAdmins();
+
+        if (!isPutRequestFresh(req, adminsConfigFromServer)) {
+            throw haltBecauseEtagDoesNotMatch("system_admins", "");
```

Comment:
This will throw an error with the following message:
`Someone has modified the configuration for system_admins ''. Please update your copy of the config with the changes and try again.`

How about just using `haltBecauseEtagDoesNotMatch()` ? it will throw a generic message saying: `Someone has modified the entity. Please update your copy with the changes and try again.`